app/components/UI/CollectibleContracts/index.test.tsx

```diff
@@ -130,4 +148,122 @@ describe('CollectibleContracts', () => {
+    const CURRENT_ACCOUNT = '0x1a';
+    const collectibleData = [
+      {
+        address: '0x72b1FDb6443338A158DeC2FbF411B71aeB157A42',
+        name: 'MyToken',
+        symbol: 'MTK',
+      },
+    ];
+    const nftItemData = [
+      {
+        address: '0x72b1FDb6443338A158DeC2FbF411B71aeB157A42',
+        description:
+          'Lil Pudgys are a collection of 22,222 randomly generated NFTs minted on Ethereum.',
+        error: 'Opensea import error',
+        favorite: false,
+        image: 'https://api.pudgypenguins.io/lil/image/11222',
+        isCurrentlyOwned: true,
+        name: 'Lil Pudgy #113',
+        standard: 'ERC721',
+        tokenId: '113',
+        tokenURI: 'https://api.pudgypenguins.io/lil/113',
+      },
+    ];
+
+    const nftItemDataUpdated = [
+      {
+        address: '0x72b1FDb6443338A158DeC2FbF411B71aeB157A42',
+        description:
+          'Lil Pudgys are a collection of 22,222 randomly generated NFTs minted on Ethereum.',
+        error: 'Opensea import error',
+        favorite: false,
+        image: 'https://api.pudgypenguins.io/lil/image/updated.png',
+        isCurrentlyOwned: true,
+        name: 'Lil Pudgy #113',
+        standard: 'ERC721',
+        tokenId: '113',
+        tokenURI: 'https://api.pudgypenguins.io/lil/113',
+      },
+    ];
+    const mockState = {
+      collectibles: {
+        favorites: {},
+      },
+      engine: {
+        backgroundState: {
+          ...initialBackgroundState,
+          NetworkController: {
+            network: '1',
+            providerConfig: {
+              ticker: 'ETH',
+              type: 'mainnet',
+              chainId: '1',
+            },
+          },
+          AccountTrackerController: {
+            accounts: { [CURRENT_ACCOUNT]: { balance: '0' } },
+          },
+          PreferencesController: {
+            displayNftMedia: true,
+            selectedAddress: CURRENT_ACCOUNT,
+            identities: {
+              [CURRENT_ACCOUNT]: {
+                address: CURRENT_ACCOUNT,
+                name: 'Account 1',
+              },
+            },
+          },
+          NftController: {
+            addNft: jest.fn(),
+            allNfts: {
+              [CURRENT_ACCOUNT]: {
+                '1': [],
+              },
+            },
+            allNftContracts: {
+              [CURRENT_ACCOUNT]: {
+                '1': [],
+              },
+            },
+          },
+        },
+      },
+    };
+
+    jest
+      .spyOn(allSelectors, 'collectiblesSelector')
+      .mockReturnValueOnce(nftItemData);
+    jest
+      .spyOn(allSelectors, 'collectibleContractsSelector')
+      .mockReturnValue(collectibleData);
+    jest
+      .spyOn(allSelectors, 'collectiblesSelector')
+      .mockReturnValueOnce(nftItemDataUpdated);
+
+    const spyOnAddNft = jest
+      .spyOn(Engine.context.NftController, 'addNft')
+      .mockImplementation(async () => undefined);
```

Comment:
Could be simplified with:
```suggestion
    const spyOnCollectibles = jest
      .spyOn(allSelectors, 'collectiblesSelector')
      .mockReturnValueOnce(nftItemData)
      .mockReturnValueOnce(nftItemDataUpdated);
    const spyOnContracts = jest
      .spyOn(allSelectors, 'collectibleContractsSelector')
      .mockReturnValue(collectibleData);
    const spyOnAddNft = jest
      .spyOn(Engine.context.NftController, 'addNft')
      .mockImplementation(async () => undefined);
```

And so, with naming the spies, you could be able to clean them up at the end of the test:
```ts
    spyOnCollectibles.mockRestore();
    spyOnContracts.mockRestore();
    spyOnAddNft.mockRestore();
```